Description: fix use after free in qlocale_unix.cpp
 This patch avoids the use of the returned variable when the destructor
 gets called.
 .
 Lisandro: this patch will certainly work within all architectures
 that Debian supports, although I don't know if it will work in QNX.
Author: Julien Cristau <julien.cristau@logilab.fr>
Bug-Debian: http://bugs.debian.org/681476
Origin: http://bugs.debian.org/681476
Forwarded: https://bugreports.qt-project.org/browse/QTBUG-26602
Reviewed-By: Lisandro Damián Nicanor Pérez Meyer <lisandro@debian.org>
Last-Update: 2012-07-20

--- a/src/corelib/tools/qlocale_unix.cpp
+++ b/src/corelib/tools/qlocale_unix.cpp
@@ -52,7 +52,7 @@
 
 QT_BEGIN_NAMESPACE
 
-static const char *getSystemLocale()
+static QByteArray getSystemLocale()
 {
 #if defined(Q_OS_QNX)
     static char buff[257];
